src/pages/Authentication/Signup.tsx

```diff
@@ -285,31 +290,67 @@ const Signup = () => {
       }
     }
     if (e.currentTarget.name === "password-check") {
-      setPasswordCheck((prevPasswordCheck) => ({
-        ...prevPasswordCheck,
-        type: prevPasswordCheck.type === "password" ? "text" : "password",
+      setPasswordConfirmation((prevPasswordConfirmation) => ({
+        ...prevPasswordConfirmation,
+        type:
+          prevPasswordConfirmation.type === "password" ? "text" : "password",
       }));
 
       if (checkEyesImgRef.current) {
         checkEyesImgRef.current.src =
-          passwordCheck.type === "password" ? visibility_off : visibility_on;
+          passwordConfirmation.type === "password"
+            ? visibility_off
+            : visibility_on;
       }
     }
   };
 
+  const handleSignup = async (e: MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+
+    if (
+      email.isValid ||
+      nickname.isValid ||
+      password.isValid ||
+      passwordConfirmation.isValid
+    ) {
+      return;
+    }
+
+    try {
+      await signup(
+        email.value,
+        nickname.value,
+        password.value,
+        passwordConfirmation.value
+      );
+
+      navigate("/login");
+    } catch (error) {
+      console.error("회원가입에 실패했습니다.", error);
+    }
+  };
+
+  useEffect(() => {
+    const token = localStorage.getItem("accessToken");
+    if (token) {
+      navigate("/");
+    }
+  }, [navigate]);
+
   useEffect(() => {
     setDisabled(
       !(
         email.isValid === false &&
         password.isValid === false &&
-        passwordCheck.isValid === false &&
+        passwordConfirmation.isValid === false &&
         nickname.isValid === false
       )
     );
```

Comment:
이 로직도 다른 부분에서 써주신 것처럼 간단하게 작성하면 좋을 것 같아요
```suggestion
setDisabled(
      email.isValid ||
      password.isValid ||
      passwordConfirmation.isValid ||
      nickname.isValid
    );
```